lib/active_merchant/billing/gateways/card_connect.rb

```diff
@@ -0,0 +1,283 @@
+        post[:email] = options[:email] if options[:email]
+      end
+
+      def add_address(post, options)
+        address = options[:billing_address] || options[:address]
+        if address
+          post[:address] = address[:address1] if address[:address1]
+          post[:city] = address[:city] if address[:city]
+          post[:region] = address[:state] if address[:state]
+          post[:country] = address[:country] if address[:country]
+          post[:postal] = address[:zip] if address[:zip]
+          post[:phone] = address[:phone] if address[:phone]
+        end
+      end
+
+      def add_money(post, money)
+        post[:amount] = amount(money)
+      end
+
+      def add_currency(post, money, options)
+        post[:currency] = (options[:currency] || currency(money))
+      end
+
+      def add_invoice(post, options)
+        post[:orderid] = options[:order_id]
+        post[:ecomind] = (options[:recurring] ? 'R' : 'E')
+      end
+
+      def add_payment(post, payment)
+        post[:name] = payment.name
+        if card_brand(payment) == 'check'
+          add_echeck(post, payment)
+        else
+          post[:account] = payment.number
+          post[:expiry] = expdate(payment)
+          post[:cvv2] = payment.verification_value
+        end
+      end
+
+      def add_echeck(post, payment)
+        post[:accttype] = 'ECHK'
+        post[:account] = payment.account_number
+        post[:bankaba] = payment.routing_number
+      end
+
+      def add_reference(post, authorization)
+        post[:retref] = authorization
+      end
+
+      def add_additional_data(post, options)
+        post[:ponumber] = options[:po_number]
+        post[:taxamnt] = options[:tax_amount] if options[:tax_amount]
+        post[:frtamnt] = options[:freight_amount] if options[:freight_amount]
+        post[:dutyamnt] = options[:duty_amount] if options[:duty_amount]
+        post[:orderdate] = options[:order_date] if options[:order_date]
+        post[:shipfromzip] = options[:ship_from_zip] if options[:ship_from_zip]
+        if (shipping_address = options[:shipping_address])
+          post[:shiptozip] = shipping_address[:zip]
+          post[:shiptocountry] = shipping_address[:country]
+        end
+        if options[:items]
+          post[:items] = []
+          options[:items].each do |item|
+            updated = {}
+            item.each_pair do |k, v|
+              updated.merge!(k.to_s.gsub(/_/, '') => v)
+            end
+            post[:items] << updated
+          end
+        end
+      end
+
+      def add_3DS(post, options)
+        post[:secureflag] = options[:secure_flag] if options[:secure_flag]
+        post[:securevalue] = options[:secure_value] if options[:secure_value]
+        post[:securexid] = options[:secure_xid] if options[:secure_xid]
+      end
+
+      def headers
+        {
+          'Authorization' => 'Basic ' + Base64.strict_encode64("#{@options[:username]}:#{@options[:password]}").strip,
```

Comment:
`Base64.strict_encode64` shouldn't be including trailing spaces, so `.strip` ought to be a no-op here; should this be for the `username:password` string?